build-std: release candidate (#4)
davidtwco/rfcs: text/0000-build-std/1-background.md

Although only `std` or `core` are added to the extern prelude automatically,
users can still write `extern crate alloc` or `extern crate test` to load them
from the sysroot.

`--extern` has a `noprelude` modifier which will allow the user to use
`--extern` to specify the location at which a crate can be found without adding
it to the extern prelude. This could allow a path for crates like `alloc` or
`test` to be provided without effecting the observable behaviour of the
language.

## Panic strategies
[background-panic-strategies]: #panic-strategies

Rust has the concept of a *panic handler*, which is a crate that is responsible
for performing a panic. There are various panic handler crates on crates.io,
such as [panic-abort] (which different from the `panic_abort` panic runtime!),
[panic-halt], [panic-itm], and [panic-semihosting]. Panic handler crates define
a function annotated with `#[panic_handler]`. There can only be one
`#[panic_handler]` in the crate graph.

`core` uses the panic handler to implement panics inserted by code generation
(e.g. arithmetic overflow or out-of-bounds access) and the `core::panic!` macro
immediately delegates to the panic handler crate.

> `std` is also a panic handler. `std`'s panic handler function and its

Reviewer: ```suggestion
`std` defines a panic handler. `std`'s panic handler function and its
```

(Not sure what "is also" meant here; core is not a panic handler).

Author: Fixed in 428ce01c
